yt_dlp/extractor/bluesky.py

```diff
@@ -0,0 +1,223 @@
+            'uploader_id': 'did:web:de1.tentacle.expert',
+            'channel_id': 'did:web:de1.tentacle.expert',
+            'uploader_url': 'https://bsky.app/profile/de1.pds.tentacle.expert',
+            'channel_url': 'https://bsky.app/profile/did:web:de1.tentacle.expert',
+            'timestamp': 1726098823,
+            'like_count': int,
+            'repost_count': int,
+            'comment_count': int,
+            'webpage_url': 'https://bsky.app/profile/de1.pds.tentacle.expert/post/3l3w4tnezek2e',
+            'tags': 'count:1',
+            'subtitles': 'count:0',
+            'comments': None,
+        },
+    }]
+
+    def _get_comments(self, meta):
+        yield from self.traverse_replies(meta, traverse_obj(meta, ('post', 'uri'), default=''))
+
+    def traverse_replies(self, thread_node, root_uri):
+        post_uri = traverse_obj(thread_node, ('post', 'uri'))
+        if post_uri != root_uri:
+            post = thread_node.get('post')
+            parent_uri = traverse_obj(post, ('record', 'reply', 'parent', 'uri'))
+            author_handle = traverse_obj(post, ('author', 'handle'))
+            author_did = traverse_obj(post, ('author', 'did'), default='')
+            yield {
+                'id': post_uri,
+                'text': traverse_obj(post, ('record', 'text')),
+                'timestamp': parse_iso8601(traverse_obj(post, ('record', 'createdAt'))),
+                'parent': 'root' if parent_uri == root_uri else parent_uri,
+                'like_count': post.get('likeCount'),
+                'author': traverse_obj(post, ('author', 'displayName')),
+                'author_id': author_did,
+                'author_thumbnail': traverse_obj(post, ('author', 'avatar'), expected_type=url_or_none),
+                'author_url': f'https://bsky.app/profile/{author_handle}',
+                'author_is_uploader': author_did in root_uri,
+            }
+        if replies := thread_node.get('replies'):
+            for reply in replies:
+                yield from self.traverse_replies(reply, root_uri)
+        if parent := thread_node.get('parent'):
+            yield from self.traverse_replies(parent, root_uri)
+
+    def _real_extract(self, url):
+        handle, video_id = self._match_valid_url(url).groups()
+        did = handle if handle.startswith('did:') else self._download_json(
+            'https://public.api.bsky.app/xrpc/com.atproto.identity.resolveHandle',
+            video_id, query={'handle': handle}).get('did')
+
+        getcomments = self.get_param('getcomments', False)
+        meta = self._download_json(
+            'https://public.api.bsky.app/xrpc/app.bsky.feed.getPostThread',
+            video_id, headers={'Content-Type': 'application/json'}, query={
+                'uri': f'at://{did}/app.bsky.feed.post/{video_id}',
+                'depth': 1000 if getcomments else 0,
+                'parentHeight': 1000 if getcomments else 0,
+            }).get('thread')
```

Comment:
Since the variable is mandatory
```suggestion
            })['thread']
```